add-ons/updating/tb78/convert-legacy.md

```diff
@@ -0,0 +1,146 @@
+
+
+## Replacing options
+
+Option windows are replaced by [settings pages](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Implement_a_settings_page) defined in `manifest.json`:
+```
+"options_ui": {
+  "page": "options.html"
+},
+```
+Instead of an XUL dialog, use the specified HTML document will be accessible to the user through the Add-on manager. From that document, all WebExtension and MailExtension APIs can be accessed in the same way as form a background script. The settings themselves should be stored using one of the new APIs to store data, such as [`storage`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage), so it may be necessary to add an experiment to migrate existing settings from nsIPrefBranch or other mechanisms not accessible through modern APIs.
+
+
+## Replacing chrome.manifest
+
+The `chrome.manifest` file is no longer supported. Many mechanisms have a more or less direct equivalent in the WebExtension world:
+* **interfaces** it should no longer be necessary to use binary interfaces, as binary components are deprecated for a long time. To access custom methods on JS-implemented classes from an experiment, use `.wrappedJSObject` to get access to the raw JS implementation.
+* **component**, **contract** [see section 'Replacing XPCOM registration' below](#replacing-xpcom-registration).
+* **category** use [`nsICategoryManager`](https://developer.mozilla.org/en/XPCOM_Interface_Reference/nsICategoryManager) through an experiment
+* **content**, **skin**, **resource** it is no longer easily possible to assign chrome://- and resource://-URLs from add-ons. However, all files of an add-on are now available through moz-extension:// URLs by default. To generate such URLs, use [`browser.runtime.getURL`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/getURL) or `extension.baseURL` from an experiment. Some WebExtension APIs also accept relative paths and generate suitable URLs automatically.
+* **locale** localization for WebExtensions is handled using the [`i18n` API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/i18n), which uses a single `messages.json` file to store translations.
+* **style** use an experiment to monitor open windows, and inject the style through that experiment
+* **overlay** [see section 'Replacing Overlays' below](#replacing-overlays).
+
+There are no direct equivalents to manifest flags, so add-ons now need to provide their own mechanisms to switch code or resources depending on the runtime environment. Relevant information is accessible through the [`runtime` API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/).
+
```

Comment:
I would move this section also way up, even before the options section. The locale stuff is one of the very first things you have to update. We have a tool, which converts the old property and what-not-files to json.

Reply:
I agree with moving it up, but it should stay below experiments (which will be drastically reduced as discussed above) as it makes direct references to experiments.

To not freak out GitHub's tools more than necessary, I'd propose to do order changes as last step before merging, though ;)